### PR TITLE
PAY-358: Better formatting for table tests in TestDiv

### PR DESCRIPTION
**Pull request by JonLim**
Wanted to use `Div()` function for properly dividing a money object into smaller segments (ie. per pay period amounts) instead of using `Mulf()` because mulf was not rounding correctly.

Getting this PR up while I investigate `Mulf`, and will make a different PR for that if necessary.

### CODE REVIEW

File: money_test.go [hunk 1]
```diff
@@ -73,39 +73,76 @@ func TestMulYen(t *testing.T) {
...
+			money2:    Money{300, "EUR"},
+			expResult: Money{667, "EUR"},
+		},
+		{
+			name:      "Rational number with repeating decimals get truncated and rounded to 0 decimals",
+			money1:    Money{2000, "JPY"},
+			money2:    Money{300, "JPY"},
+			expResult: Money{7, "JPY"},
+		},
+		// PAY-355: three tests below are testing the workaround for finding per pay period amounts!
```
Comment (reillywatson) [inline money_test.go]: I don't think this comment makes sense in the context of this repo!

Comment (JonLim) [reply money_test.go]: You are right - copypasta, my mistake! Will update.

**Comment by JonLim** [comment]
@reillywatson Found out where the root cause might be!

At the end of `Mulf()`, we return this:

```go
func (m Money) Mulf(f float64) Money {
    // ...
    return Money{C: m.C, M: Rnd(r, float64(i)/guardf-float64(r))}
}
```

Which leads to this method:
```go
func Rnd(r int64, trunc float64) int64 {
	if trunc > 0 {
		if trunc >= Round {
			r++
		}
	} else {
		if trunc < Roundn {
			r--
		}
	}
	return r
}
```

And in this particular case, we're trying to multiply `M: 22668` with `1/24` (for our per pay period thing), which when run through the `Rnd()` function, will return an unmodified `r` value. This is because with our floating point math, `trunc` is `0.48488799999995535`, which is not larger than `Round` constant (`0.5`), and so will not round upward.

Didn't want to make any crazy changes to this package because of how many things might rely on `Mulf` or `Rnd`, but did you have any thoughts around this?

**Comment by reillywatson** [comment]
Baaaah, that looks bad. Check out this even worse thing! If I replace the Mulf() implementation with this totally-reasonable-looking one:
```go
func (m Money) Mulf(f float64) Money {
	return m.Mul(MakeMoney(m.C, f))
}
```
I then get 9.06 for the above multiply, rather than the 9.45 you'd expect (or the 9.44 we currently return).

**Pull request merged**